extensions/amp-3q-player/0.1/amp-3q-player.js

```diff
@@ -86,17 +83,42 @@ class Amp3QPlayer extends AMP.BaseElement {
     Services.videoManagerForDoc(el).register(this);
   }
 
-  /** @override */
-  layoutCallback() {
-    const iframe = createFrameFor(
-      this,
-      'https://playout.3qsdn.com/' +
+  /** private */
+  generateIframeSrc() {
+    const explicitParamsAttributes = ['key', 'timestamp', 'controls'];
+
+    let iframeSrc = 'https://playout.3qsdn.com/';
+
+    if (this.element.getAttribute(`data-datasource`)) {
+      iframeSrc +=
+        'config_by_metadata/' +
+        this.element.getAttribute(`data-project`) +
+        '/' +
+        this.element.getAttribute(`data-datafield`) +
+        '/' +
         encodeURIComponent(dev().assertString(this.dataId)) +
         // Autoplay is handled by VideoManager
-        '?autoplay=false&amp=true'
-    );
+        '?autoplay=false&amp=true';
+    } else {
+      iframeSrc +=
+        this.dataId +
+        // Autoplay is handled by VideoManager
+        '?autoplay=false&amp=true';
```

Comment:
Nit: Add this pair of params once when returning.

I would also prefer if you factor out a `getUrlPath()` method that returns the path value rather than modifying a string.

Reply:
done!